### DISCUSSION TITLE
environment variable not loaded from shell

### DISCUSSION THREAD

**Discussion by brookmg** [prisma/prisma#7901] [Q&A] [answered]
Why does Prisma insist on having .env file present when the variables are already exported by the shell?

**Comment by ryands17** [comment, marked as answer]
@brookmg 👋 

The `.env` isn't required if the variable is in the shell. What's the error you're facing?

**Reply by brookmg** [reply]
I'm not exactly sure. I exported some variables like `DATABASE_URL` through `.profile` on an external VPS. I was using Github actions self-runners to deploy what was committed on `main` and it basically removes everything in the working directory before cloning the repo again and running tests. Therefore `.env` file wouldn't be ideal and I would need to push it to the version control to make it appear every time. I am running the node project with pm2 and it keeps saying `DATABASE_ENV` can't be found.

**Reply by brookmg** [reply]
The `.schema` file is similar to the Prisma example. URL is fetched from `env('DATABASE_URL')` if that's the issue

**Reply by ryands17** [reply]
Does echoing the variable give the correct result in the Github runner? If not, then there's a problem setting up the env variable in `.profile`. Currently I have setup the variable on my local system and it works fine.

**Reply by ryands17** [reply]
@brookmg Was the above issue solved? If so, then it would be great if you could mark this discussion as answered 🙂

**Reply by brookmg** [reply]
Ow yes it is, thanks @ryands17

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
